common/djangoapps/util/db.py

```diff
@@ -4,44 +4,119 @@
 from functools import wraps
 import random
 
-from django.db import connection, transaction
+from django.db import DEFAULT_DB_ALIAS, DatabaseError, Error, transaction
 
 
 MYSQL_MAX_INT = (2 ** 31) - 1
 
 
-def commit_on_success_with_read_committed(func):  # pylint: disable=invalid-name
+class CommitOnSuccessManager(object):
```

Comment:
Right, though I would add that in case of nested `CommitOnSuccessManager`s, it may also happen that only part of the outer block's queries get committed. I linked this in the email and explains the difference between the two very nicely: http://stackoverflow.com/questions/21861207/is-transaction-atomic-same-as-transaction-commit-on-success/24108491#24108491